plugins/opener/src/open.rs

```diff
@@ -0,0 +1,166 @@
+    Open,
+    /// Use the `start` program.
+    Start,
+    /// Use the `xdg-open` program.
+    XdgOpen,
+    /// Use the `gio` program.
+    Gio,
+    /// Use the `gnome-open` program.
+    GnomeOpen,
+    /// Use the `kde-open` program.
+    KdeOpen,
+    /// Use the `wslview` program.
+    WslView,
+    /// Use the `Firefox` program.
+    Firefox,
+    /// Use the `Google Chrome` program.
+    Chrome,
+    /// Use the `Chromium` program.
+    Chromium,
+    /// Use the `Safari` program.
+    Safari,
+}
+
+impl Display for Program {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Open => "open",
+                Self::Start => "start",
+                Self::XdgOpen => "xdg-open",
+                Self::Gio => "gio",
+                Self::GnomeOpen => "gnome-open",
+                Self::KdeOpen => "kde-open",
+                Self::WslView => "wslview",
+                Self::Firefox => "firefox",
+                Self::Chrome => "chrome",
+                Self::Chromium => "chromium",
+                Self::Safari => "safari",
+            }
+        )
+    }
+}
+
+impl FromStr for Program {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let p = match s.to_lowercase().as_str() {
+            "open" => Self::Open,
+            "start" => Self::Start,
+            "xdg-open" => Self::XdgOpen,
+            "gio" => Self::Gio,
+            "gnome-open" => Self::GnomeOpen,
+            "kde-open" => Self::KdeOpen,
+            "wslview" => Self::WslView,
+            "firefox" => Self::Firefox,
+            "chrome" | "google chrome" => Self::Chrome,
+            "chromium" => Self::Chromium,
+            "safari" => Self::Safari,
+            _ => return Err(crate::Error::UnknownProgramName(s.to_string())),
+        };
+        Ok(p)
+    }
+}
+
+impl<'de> Deserialize<'de> for Program {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Program::from_str(&s).map_err(|e| serde::de::Error::custom(e.to_string()))
+    }
+}
+
+impl Program {
+    pub(crate) fn name(self) -> &'static str {
+        match self {
+            Self::Open => "open",
+            Self::Start => "start",
+            Self::XdgOpen => "xdg-open",
+            Self::Gio => "gio",
+            Self::GnomeOpen => "gnome-open",
+            Self::KdeOpen => "kde-open",
+            Self::WslView => "wslview",
+
+            #[cfg(target_os = "macos")]
+            Self::Firefox => "Firefox",
+            #[cfg(not(target_os = "macos"))]
+            Self::Firefox => "firefox",
+
+            #[cfg(target_os = "macos")]
+            Self::Chrome => "Google Chrome",
+            #[cfg(not(target_os = "macos"))]
+            Self::Chrome => "google-chrome",
```

Comment:
This doesn't seem to work on Windows, it should be `chrome`

Reply:
I think I will change to remove this enum altogether.

Reply:
This enum seems to support some commonly used programs cross-platforms, I agree we remove it for now since obviously it doesn't do what it supposed to do right now, we can bring it back in the future when we actually made it work